### PR TITLE
Bump native provider version to 5.28.0

### PR DESCRIPTION
**Pull request by turkenf**
### Description of your changes

Upgraded Terraform provider version to `v5.28.0`

Fixes https://github.com/crossplane-contrib/provider-upjet-gcp/issues/517 

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- ~[ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.~

#### Breaking CRD schema changes

- `package/crds/cloudrun.gcp.upbound.io_v2services.yaml` => with uptest
    - `v1beta1`: https://github.com/crossplane-contrib/provider-upjet-gcp/actions/runs/9766550701
    - `v1beta2`: https://github.com/crossplane-contrib/provider-upjet-gcp/actions/runs/9776841576

- `package/crds/kms.gcp.upbound.io_cryptokeyversions.yaml` => manually
```
NAME                                                     SYNCED   READY   EXTERNAL-NAME
                                                          AGE
cryptokeyversion.kms.gcp.upbound.io/key-version-fttest   True     True    projects/official-provider-testing/locations/global/keyRings/keyring-crypto-key-version-fttest/c
ryptoKeys/crypto-key-version-fttest/cryptoKeyVersions/2   9m57s
```

#### Successful random tests

- `examples/cloudplatform/v1beta1/serviceaccount.yaml` => https://github.com/crossplane-contrib/provider-upjet-gcp/actions/runs/9766576123
- `examples/storage/v1beta2/bucket.yaml` => https://github.com/crossplane-contrib/provider-upjet-gcp/actions/runs/9776657587
- `examples/spanner/v1beta2/database.yaml` => https://github.com/crossplane-contrib/provider-upjet-gcp/actions/runs/9776794412
- `examples/dns/v1beta2/policy.yaml` => https://github.com/crossplane-contrib/provider-upjet-gcp/actions/runs/9776857493
- `examples/container/v1beta2/cluster.yaml` => https://github.com/crossplane-contrib/provider-upjet-gcp/actions/runs/9776864525
- `examples/compute/v1beta2/router.yaml` => https://github.com/crossplane-contrib/provider-upjet-gcp/actions/runs/9776873949
- `examples/compute/v1beta1/firewall.yaml` => https://github.com/crossplane-contrib/provider-upjet-gcp/actions/runs/9776889909

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by turkenf** [comment]
/test-examples="examples/cloudrun/v1beta1/v2service.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/cloudplatform/v1beta1/serviceaccount.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/spanner/v1beta2/database.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/cloudrun/v1beta2/v2service.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/dns/v1beta2/policy.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/container/v1beta2/cluster.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/compute/v1beta2/router.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/compute/v1beta1/firewall.yaml"

File: package/crds/accesscontextmanager.gcp.upbound.io_serviceperimeterresources.yaml [hunk 1]
```diff
@@ -35,8 +35,7 @@ spec:
     schema:
       openAPIV3Schema:
         description: ServicePerimeterResource is the Schema for the ServicePerimeterResources
-          API. Allows configuring a single GCP resource that should be inside of a
-          service perimeter.
+          API. Allows configuring a single GCP resource that should be inside the
```
Comment (mergenci) [inline package/crds/accesscontextmanager.gcp.upbound.io_serviceperimeterresources.yaml:38]: Turns out code generation wasn't able to properly handle an HTML code tag in [the documentation](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/access_context_manager_service_perimeter_resource). IMHO, this is not crucial, but let's be aware of the issue.

**Review by mergenci — APPROVED** [review]
Thanks @turkenf. LGTM.

File: package/crds/accesscontextmanager.gcp.upbound.io_serviceperimeterresources.yaml (hunk 1 above)
Comment (turkenf) [reply package/crds/accesscontextmanager.gcp.upbound.io_serviceperimeterresources.yaml:38]: I fixed the description by adding a manual configuration, but I agree that these should be handled during code generation: [c75257f](https://github.com/crossplane-contrib/provider-upjet-gcp/pull/553/commits/c75257f49e188cf568dbcd548f1883388dbb5faf)[c75257f](https://github.com/crossplane-contrib/provider-upjet-gcp/pull/553/commits/c75257f49e188cf568dbcd548f1883388dbb5faf)[c75257f](https://github.com/crossplane-contrib/provider-upjet-gcp/pull/553/commits/c75257f49e188cf568dbcd548f1883388dbb5faf)

**Comment by turkenf** [comment]
/test-examples="examples/cloudrun/v1beta1/v2service.yaml"

**Pull request merged**